### PR TITLE
Add a controller-based default for window titles

### PR DESCRIPTION
**Pull request by Vin0uz**
It's not completely answering #1020 , which btw is not totally true, some pages have a `content_for` defined already :D (stats or relatives for example, but I am sure some others have too ;))

Two commits in this diff : 
- Adding a leading space to fix `StatistiquesDu`
- Adding a translation for titles, based on controller names. 

Some are overriden, but I kept a translation as a fallback. 

I am really not sure about the translations, so please adjust if needed 🙏

### CODE REVIEW

File: app/views/stats/index.html.slim [hunk 1]
```diff
@@ -3,7 +3,7 @@
     | Statistiques
     = " pour le #{@departement}" if @departement.present?
     br
-    small Du #{l(Stat.default_date_range.first.to_date)} au #{l(Stat.default_date_range.last.to_date)}
+    small< Du #{l(Stat.default_date_range.first.to_date)} au #{l(Stat.default_date_range.last.to_date)}
```
Comment (adipasquale) [inline app/views/stats/index.html.slim:6]: just learnt about this `<` to prepend a whitespace, I didn't know, thanks!

**Review by adipasquale — APPROVED** [review]

**Comment by adipasquale** [comment]
Thank you for this PR @Vin0uz it's great !

I see some things that we could improve in the future: 

- suffix all titles with `- RDV Solidarités` (or prefix, but I think it makes less sense, you should get the most valuable info first). Currently, most defined content_for title don't include RDV-Solidarités as a suffix, which makes sense, this should probably be appended in the layout.
- check that there are titles on the user side too
- remove I18n keys that you added that are not actually used, I have a feeling that most are unused actually. `title` content_for seems to be defined in most templates. 

However this is a great first step and a nice improvement compared to now.

**Comment by adipasquale** [comment]
about the spec, I think it's fine not to overcomplicate / not to test at all for such non-core presentation concerns

**Comment by Vin0uz** [comment]
> I see some things that we could improve in the future:
> 
> * suffix all titles with `- RDV Solidarités` (or prefix, but I think it makes less sense, you should get the most valuable info first). Currently, most defined content_for title don't include RDV-Solidarités as a suffix, which makes sense, this should probably be appended in the layout.
> * check that there are titles on the user side too
> * remove I18n keys that you added that are not actually used, I have a feeling that most are unused actually. `title` content_for seems to be defined in most templates.
> 
> However this is a great first step and a nice improvement compared to now.

Would you prefer to merge this diff and open another one with improvements, or just edit/pile this one ?

**Comment by adipasquale** [comment]
Ideally @Vin0uz I'd say that suffixing with " - RDV-Solidarités" and removing unused i18n keys should be done in this PR, and that the user side can be pushed back in an isolated PR.

However I'm completely fine with merging this PR as it is and then working maybe someday on these 3 points, or not and leaving it as is, it's much better than today already!

**Comment by Vin0uz** [comment]
> Ideally @Vin0uz I'd say that suffixing with " - RDV-Solidarités" and removing unused i18n keys should be done in this PR, and that the user side can be pushed back in an isolated PR.
> 
> However I'm completely fine with merging this PR as it is and then working maybe someday on these 3 points, or not and leaving it as is, it's much better than today already!

Turned out only few pages were missing `content_for :title`, sorry guys, I should have checked better before pushing 🙈

**Review by adipasquale — CHANGES_REQUESTED** [review]
🙇 thanks for removing the redundant i18n keys

there is just one minor change that's necessary now, I had not seen this coming. On the stats page, the `content_for :title` contains HTML tags h1, br. We therefore cannot use it as is in the window title. it currently looks like `<h1>Statistiques <br /> ....` . That's also why the specs fail currently.

You can simply use the `strip_tags` Rails helper which does exactly what we want here, and it fixes the specs :

```
diff --git a/app/views/common/_meta.html.slim b/app/views/common/_meta.html.slim
index 4f59deb1..9ffd88f3 100644
--- a/app/views/common/_meta.html.slim
+++ b/app/views/common/_meta.html.slim
@@ -2,7 +2,7 @@ meta name="viewport" content="width=device-width, initial-scale=1.0"
 meta content="text/html; charset=UTF-8" http-equiv="Content-Type"
 title
   - if content_for? :title
-    = "#{content_for(:title)} -  RDV Solidarités"
+    = "#{strip_tags(content_for(:title))} -  RDV Solidarités"
   - else
     | RDV Solidarités
```

I cannot apply this patch myself to your PR, so could you either give me permission to do so (I'm not sure but I think you have to add me as a collaborator to your forked repository of rdv-solidarites) or apply it ? 

Again, thank you and thanks for bearing with my requests 🙇

**Comment by Vin0uz** [comment]
> 🙇 thanks for removing the redundant i18n keys
> 
> there is just one minor change that's necessary now, I had not seen this coming. On the stats page, the `content_for :title` contains HTML tags h1, br. We therefore cannot use it as is in the window title. it currently looks like `<h1>Statistiques <br /> ....` . That's also why the specs fail currently.
> 
> You can simply use the `strip_tags` Rails helper which does exactly what we want here, and it fixes the specs :
> 
> ```
> diff --git a/app/views/common/_meta.html.slim b/app/views/common/_meta.html.slim
> index 4f59deb1..9ffd88f3 100644
> --- a/app/views/common/_meta.html.slim
> +++ b/app/views/common/_meta.html.slim
> @@ -2,7 +2,7 @@ meta name="viewport" content="width=device-width, initial-scale=1.0"
>  meta content="text/html; charset=UTF-8" http-equiv="Content-Type"
>  title
>    - if content_for? :title
> -    = "#{content_for(:title)} -  RDV Solidarités"
> +    = "#{strip_tags(content_for(:title))} -  RDV Solidarités"
>    - else
>      | RDV Solidarités
> ```
> 
> I cannot apply this patch myself to your PR, so could you either give me permission to do so (I'm not sure but I think you have to add me as a collaborator to your forked repository of rdv-solidarites) or apply it ?
> 
> Again, thank you and thanks for bearing with my requests 🙇

Didn't see that coming indeed 😮  nice one ! It seems that tags are present in other places (I found `admin/organisations/stats/index`, `admin/stats/index`, `static_pages/contact`, `static_pages/mds`, `stats/index` so far).

I have a feeling that `:title` means different thing around the app : title displayed on the page, and html `<title>`. So those two things overlap each other, in layouts and views. 
The `strip_tags` will prevent seeing HTML tags but would probably cause missing spaces between elements. Injecting space before HTML tags, and then strip them is a possiblity but feels dirty a bit 🙈  

Quick question about the repo's style : Do you prefer people to edit their commits with changes, or to pile commits on after the other to keep trace of the whole process ?

**Comment by adipasquale** [comment]
@Vin0uz good catch I had not seen that there were other usages of html tags in titles.

I'm guessing if we wanted a 'perfect' solution we could add yet another `content_for(:window_title)` that would specifically contain the window title, not the page-displayed title. We could imagine that there is a fallback mechanism : in 90% of the cases it's fine to use the existing `title` for both the page title and the window title, but we allow to specifically use a `window_title` in cases where there are HTML tags used in the page title.

In my opinion this might be slightly over-engineered for now, and unless we have seen an actual problem with using `strip_tags` I think we're better of leaving it as is for now, with a single `title` content_for. This is completely open for discussion. 

What we tend to do in PRs is :
- you're free to create as many commits as you'd like
- try not to rewrite history using `git commit --amend` or `git rebase master` and such once you've asked for a review, so that reviewers can follow the process
- when we merge the PR we usually squash all commits together unless there are individual commits that were thought to have a lot of meaning separately and could work on their own as proper releases, this does not happen often.

**Comment by Vin0uz** [comment]
@adipasquale I agree that having another keyword is probably too much. The pages that I listed earlier will have "weird" page titles (ex for contact: `ContactDernière mise à jour : le 26 octobre 2020 - RDV Solidarités`)

 It can be considered "ok" to proceed, or hold 'til everything ✨ 
 That's your call ^^'

**Comment by adipasquale** [comment]
good catch @Vin0uz , I think we'll simply remove this time information from the contact page title, it doesn't make so much sense for it to be there. I'll merge this PR now, again thank you a lot for your work on this 🙇

**Review by adipasquale — APPROVED** [review]

**Pull request merged**